internal/hcsoci/hcsdoc_wcow.go

```diff
@@ -23,6 +23,61 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// createMountsConfig generates the arrays of MappedDirectory or MappedPipe requests
+// based on the mounts described in the container spec. Handles the mounts for both
+// v1 and v2 schema.
+func createMountsConfig(ctx context.Context, coi *createOptionsInternal) ([]schema1.MappedDir, []schema1.MappedPipe, []hcsschema.MappedDirectory, []hcsschema.MappedPipe, error) {
```

Comment:
it kind of seems like maybe we should be returning some kind of struct here?

Reply:
might also be worth refactoring some of the code here :)

Reply:
What do you mean by refactoring? Can you explain more?